### PR TITLE
Added best practice recommendations for using DFS with Octopus Server

### PR DESCRIPTION
**Pull request by gian-lorenzetto**
Clarified DFS usage with Octopus Server, for both single and multi-node configurations.

- Added a new best practices page for DFS configuration
- Removed DFS as an example from the installation guidelines
- Updated the DFS link from the HA design page to the new DFS recommendations information

### CODE REVIEW

**Comment by gian-lorenzetto** [comment]
[sc-7742]

File: docs/administration/high-availability/design/octopus-for-high-availability-on-premises.md [hunk 1]
```diff
@@ -42,7 +42,8 @@ Since each of the Octopus Server nodes will need access to the database, we reco
 
 The simplest way to provide shared storage, assuming the Octopus Server nodes are part of the same Active Directory domain, is by creating a file share that each of the Octopus Server nodes can access. Of course, this assumes that the underlying directory is reliable, such as in a RAID array.
 
-A better alternative is [Microsoft DFS](https://en.wikipedia.org/wiki/Distributed_File_System_(Microsoft)) or a SAN.
+An alternative is [Microsoft DFS](https://en.wikipedia.org/wiki/Distributed_File_System_(Microsoft)). If using Microsoft DFS for the shared storage, it must be [configured specifically for use with Ocotpus Deploy](docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md).
```
Comment (jbristowe) [inline docs/administration/high-availability/design/octopus-for-high-availability-on-premises.md:45]:
```suggestion
An alternative is [Microsoft DFS](https://en.wikipedia.org/wiki/Distributed_File_System_(Microsoft)). If using Microsoft DFS for the shared storage, it must be [configured specifically for use with Octopus Deploy](docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md).
```

File: docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Below are recommendations and more details on:
+
+- [Configuring DFS with a Single Octopus Server](#Configuring-Octopus-with-a-Single-Octopus-Server)
+- [Configuring DFS with a Multi-Node Octopus Server cluster (Octopus HA)](#Configuring-DFS-with-a-Multi-Node-Octopus-Server-cluster-(Octopus-HA))
+- [DFS for Redundancy (Disaster Recover)](#DFS-for-Redundancy-(Disaster-Recover))
+
+## Configuring DFS with a Single Octopus Server
+
+For a single node Octopus server using DFS for file storage it is imperative that the node is **configured to use a specific DFS Replica and not the DFS Namespace Root**. Despite there being no contention between nodes in the single node configuration, there is still the DFS location transparency, which will cause unpredictable behavior when the node is directed to a different replica.
```
Comment (jbristowe) [inline docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md:22]:
```suggestion
For a single node Octopus Server using DFS for file storage it is imperative that the node is **configured to use a specific DFS Replica and not the DFS Namespace Root**. Despite there being no contention between nodes in the single node configuration, there is still the DFS location transparency, which will cause unpredictable behavior when the node is directed to a different replica.
```

File: docs/administration/high-availability/design/octopus-for-high-availability-on-premises.md (hunk 1 above)
Comment (gian-lorenzetto) [reply docs/administration/high-availability/design/octopus-for-high-availability-on-premises.md:45]: Good pick up! Thanks

File: docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md (hunk 2 above)
Comment (gian-lorenzetto) [reply docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md:22]: Fixed

**Comment by gian-lorenzetto** [comment]
Updated the single- and multi-node images, just waiting on the redundancy image before merging.

File: docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+---
+title: Configuring Microsoft DFS with Octopus Server
+description: Guidelines and recommendations for configuring Microsoft DFS as the shared file system for Octopus Deploy.
+position: 120
+hideInThisSection: true
+---
+
+Microsoft DFS is a [distributed file system](https://en.wikipedia.org/wiki/Clustered_file_system#Distributed_file_systems), which is important to consider when configuring Octopus Deploy with a DFS file share. Octopus Server makes specific assumptions about the performance and consistency of the file system when accessing log files, performing log retention, storing deployment packages and other deployment artifacts and temporary storage when communicating with Tentacles.
```
Comment (harrisonmeister) [inline docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md:8]:
```suggestion
Microsoft DFS is a [distributed file system](https://en.wikipedia.org/wiki/Clustered_file_system#Distributed_file_systems), which is important to consider when configuring Octopus Deploy with a DFS file share. Octopus Server makes specific assumptions about the performance and consistency of the file system when accessing log files, performing log retention, storing deployment packages and other deployment artifacts, and temporary storage when communicating with Tentacles.
```

File: docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+**DFS in the standard configuration (ie, accessed through a DFS Namespace Root) is _not_ suitable for use as a shared file store with Octopus Deploy.**
+
+Operating Octopus Deploy with the non-recommended DFS configuration will likely result in intermittent and potentially significant issues.
```
Comment (harrisonmeister) [inline docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md:12]: Suggest adding a call-out (warning), as this seems pretty important to highlight
```suggestion
:::warning
**DFS in the standard configuration (i.e., accessed through a DFS Namespace Root) is _not_ suitable for use as a shared file store with Octopus Deploy.**

Operating Octopus Deploy with the non-recommended DFS configuration will likely result in intermittent and potentially significant issues.
:::

File: docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+- [Configuring DFS with a Single Octopus Server](#Configuring-Octopus-with-a-Single-Octopus-Server)
+- [Configuring DFS with a Multi-Node Octopus Server cluster (Octopus HA)](#Configuring-DFS-with-a-Multi-Node-Octopus-Server-cluster-(Octopus-HA))
+- [DFS for Redundancy (Disaster Recover)](#DFS-for-Redundancy-(Disaster-Recover))
+
+## Configuring DFS with a Single Octopus Server
+
+For a single node Octopus Server using DFS for file storage it is imperative that the node is **configured to use a specific DFS Replica and not the DFS Namespace Root**. Despite there being no contention between nodes in the single node configuration, there is still the DFS location transparency, which will cause unpredictable behavior when the node is directed to a different replica.
+
+In the diagram the single node is configured to use the replica `\\SVR_ONE\public` as the DFS file share and not the namespace root (`\\Contoso\public`).
```
Comment (harrisonmeister) [inline docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md:24]:
```suggestion
For a single-node Octopus Server using DFS for file storage, the node must be **configured to use a specific DFS Replica and not the DFS Namespace Root**. Despite no contention between nodes in the single-node configuration, there is still the DFS location transparency, which will cause unpredictable behavior when the node is directed to a different replica.

In the diagram, the single node is configured to use the replica `\\SVR_ONE\public` as the DFS file share and not the namespace root (`\\Contoso\public`). 
```

File: docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+In the diagram the single node is configured to use the replica `\\SVR_ONE\public` as the DFS file share and not the namespace root (`\\Contoso\public`). 
+
+![A single Octopus Deploy node with DFS shared storage](images/single-node-od-with-dfs.png "width=500")
+
+## Configuring DFS with a Multi-Node Octopus Server cluster (Octopus HA)
+
+For a multi-node Octopus cluster using DFS for file storage it is imperative that **_all_ nodes in the cluster are configured to use the same DFS Replica and not the DFS Namespace Root**. Both using the namespace root, or using different replicas for different Octopus nodes will cause unpredictable behavior.
+
+In the diagram below each node in the cluster is configured to use the same replica (`\\SVR_ONE\public`) as the DFS file share and not the namespace root (`\\Contoso\public`).
```
Comment (harrisonmeister) [inline docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md:32]:
```suggestion
For a multi-node Octopus cluster using DFS for file storage, it is imperative that **_all_ nodes in the cluster are configured to use the same DFS Replica and not the DFS Namespace Root**. Both using the namespace root or using different replicas for different Octopus nodes will cause unpredictable behavior.

In the diagram below each node in the cluster is configured to use the same replica (`\\SVR_ONE\public`) as the DFS file share and not the namespace root (`\\Contoso\public`). 
```

File: docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+## DFS for Redundancy (Disaster Recovery)
+
+DFS can still be used for redundancy and disaster recovery as normal.
+
+In the event that the replica that OD is configured to use becomes unavailable, simply changing the configuration to another replica in the DFS Namespace with the same target folders is sufficient to restore service.
+
+OD does not need to be restarted in this scenario. Customers can either do this manually, or can automate.
+
+In the simplified diagram below, when an outage at DFS Replica `\\SVR_ONE\Public` occurs, by re-configuring each Octopus node to use a different replica (ensuring all nodes are re-configured to the same replica) customers can still take advantage of the redundancy within DFS.
```
Comment (harrisonmeister) [inline docs/getting-started/best-practices/configuring-microsoft-dfs-with-octopus-server.md:44]:
```suggestion
DFS can still be used for redundancy and disaster recovery, as usual.

If the replica that Octopus is configured to use becomes unavailable, simply changing the configuration to another replica in the DFS Namespace with the same target folders is sufficient to restore service.

Octopus does not need to be restarted in this scenario. Customers can either do this manually or can automate it.

In the simplified diagram below, when an outage at DFS Replica `\\SVR_ONE\Public` occurs, by re-configuring each Octopus node to use a different replica (ensuring all nodes are re-configured to the same replica), customers can still take advantage of the redundancy within DFS.
```

**Review by harrisonmeister — COMMENTED** [review]
Made some minor (Grammarly inspired) suggestions, and one for a warning callout

**Comment by gian-lorenzetto** [comment]
> Made some minor (Grammarly inspired) suggestions, and one for a warning callout

Thanks @harrisonmeister I've accepted your changes. This is good to go from my side now, thanks for the reviews @harrisonmeister and @jbristowe.

**Review by harrisonmeister — APPROVED** [review]
LGTM

**Pull request merged**